src/solid/Virtualizer.tsx

```diff
@@ -280,9 +284,36 @@ export const Virtualizer = <T,>(props: VirtualizerProps<T>): JSX.Element => {
     })
   );
 
-  const dataSlice = createMemo<T[]>(() => {
+  const dataSlice = createMemo<{ item: T, index: number }[]>(() => {
     const [start, end] = range();
-    return end >= 0 ? props.data.slice(start, end + 1) : [];
+    const items: { item: T, index: number }[] = [];
+    
+    for (let i = start, j = end; i <= j; i++) {
+        const item = props.data[i];
+        if (item === undefined) continue;
+        
+        items.push({ item, index: i });
```

Comment:
should I instead declare that it is not undefined?
```suggestion
        const item = props.data[i]!;
        items.push({ item, index: i });
```